activerecord/lib/active_record/associations.rb

```diff
@@ -1289,6 +1289,7 @@ module ClassMethods
         #   similar callbacks may affect the <tt>:dependent</tt> behavior, and the
         #   <tt>:dependent</tt> behavior may affect other callbacks.
         #
+        #   * <tt>nil</tt> do nothing (default).
         #   * <tt>:destroy</tt> causes all the associated objects to also be destroyed.
         #   * <tt>:delete_all</tt> causes all the associated objects to be deleted directly from the database (so callbacks will not be executed).
         #   * <tt>:nullify</tt> causes the foreign keys to be set to +NULL+. Polymorphic type will also be nullified
```

Comment:
Note that originally this list is documented the `valid_dependent_options` without `nil`.
All falsy values (not only `nil` but also `false`) are ignored `valid_dependent_options` check.

https://github.com/rails/rails/blob/2c008d9f6311d92b3660193285230505e74a114f/activerecord/lib/active_record/associations/builder/has_many.rb#L16-L18

I know the purpose of this PR is to allow `dependent: nil` which is your preference in rubocop-rails https://github.com/rubocop-hq/rubocop-rails/issues/147.

If someone want to allow `dependent: false` too in rubocop-rails, should we document `dependent: false` too?

Reply:
No. `false` and `nil` are the same for the purpose of this feature.